.github/workflows/build.yml

```
@@ -14,14 +14,18 @@ jobs:
           update: true
           path-type: inherit
       - name: Check out code
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
         with:
           fetch-depth: 0
       - name: Install golang
-        uses: actions/setup-go@v2
+        uses: actions/setup-go@v3
```

```suggestion
        uses: actions/setup-go@v3
        with:
          go-version: "1.17.9"
```

should this be set? otherwise, it will find `1.17` from the `go.mod` file and get the latest patch version (which is 1.17.13).

---

I think we should upgrade to 1.17.13 everywhere, it's true — we are currently stuck on 1.17.9 in all the various CI and build jobs.

---

can we at least do 1.17, so it doesn't go to 1.18?

---

Sure, but for that I would also want to set `scripts/get_golang_version.sh:BUILD=1.17.9` and the other references to 1.17.9 in .circleci/config.yml

---

why do you think it will find 1.17 in go.mod? i don't see it anywhere in the description. maybe we should at least point it to go.mod? (To avoid going to 1.18 prematurely)
https://github.com/actions/setup-go#getting-go-version-from-the-gomod-file

---

you are also removing 1.17.9 from cache, so this is not outside of the scope

---

ohh I didn't realize I'm removing the with: 1.17.9 line for setup-go ... will fix

---

I updated to 1.17.13 everywhere!

---

let's still specify the version. b/c it's windows there probably wouldn't be much downside to having it roll, but I feel like having this explicit would be easier when doing an analysis of during an upgrade.